src/frontend/src/service/api-service.ts

```diff
@@ -15,14 +15,17 @@ export type Request = {
   withCredentials?: boolean;
 };
 
-// The dispatch function is needed in order to send requests to the Request Middleware, which
-// handles authentication.
+const REQUEST = 'REQUEST';
+
 export default function apiService(
   dispatch: Function,
   request: Request
 ): AxiosPromise {
-  // The Action dispatched here is handled in the Request Middleware, so you can call
-  // apiService(...).then(...).catch(...) like a normal Axios request.
-  let result = dispatch({ type: REQUEST, request } as RequestAction);
-  return result.promise;
+  dispatch({ type: REQUEST, request });
```

Comment:
I think we should get rid of this dispatch (and the REQUEST constant above) because AFAIK as of this PR this action does nothing,